src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy

```diff
@@ -1321,6 +1322,39 @@ class FlowCrudSpec extends HealthCheckSpecification {
         !error && flowHelperV2.deleteFlow(flow.flowId)
     }
 
+    @Tidy
+    def "Unable to update to a flow with incorrect maxLatency input - #title"() {
+        given: "A flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelperV2.randomFlow(srcSwitch, dstSwitch)
+        flowHelperV2.addFlow(flow)
+
+        when: "Try to update the flow "
+        def flowInfo = northboundV2.getFlow(flow.flowId)
+        flowInfo = flowInfo.tap {it.maxLatency = maxLatencyValue
+            it.maxLatencyTier2 = maxLatencyTier2}
+        northboundV2.updateFlow(flowInfo.flowId,
+                flowHelperV2.toRequest(flowInfo))
+
+        then: "Bad Request response is returned"
+        def error = thrown(HttpClientErrorException)
+        error.statusCode == HttpStatus.BAD_REQUEST
+        def errorDetails = error.responseBodyAsString.to(MessageError)
+        errorDetails.errorMessage == message
+        errorDetails.errorDescription == errorDescr
+
+        cleanup: "Remove the flow"
+        flowHelperV2.deleteFlow(flow.flowId)
+        where:
+        maxLatencyValue|maxLatencyTier2|message|errorDescr|title
```

Comment:
Even if we can't fully test max latency validation on the unit level, we shouldn't run three functional tests around it. Have one functional test on any case and verify the rest validation on unit level.